src/mesh/voxel_cloud.rs

```diff
@@ -448,6 +448,165 @@ impl ScalarField {
+        {
+            return None;
+        }
+
+        let marching_cubes_lookup_table = marching_cubes_lookup_table();
+
+        // A collection of mesh patches defining an outer envelope of volumes
+        // stored in the scalar field
+        let mut marching_cubes: Vec<Mesh> = Vec::new();
+
+        let block_end = self.block_end();
+
+        // Iterate through the scalar field. Consider 8 neighboring voxels to be
+        // corners of the test cube. Start iterating at position [-1, -1, -1]
+        // relative to the `block_start` and finish at [1, 1, 1] relative to the
+        // `block_end` to catch the block boundaries.
+        for z in (self.block_start.z - 1)..=block_end.z {
+            for y in (self.block_start.y - 1)..=block_end.y {
+                for x in (self.block_start.x - 1)..=block_end.x {
+                    //                 v4_______e4_____________v5
+                    //                  /|                    /|
+                    //                 / |                   / |
+                    //              e7/  |                e5/  |
+                    //               /___|______e6_________/   |
+                    //            v7|    |                 |v6 |e9
+                    //              |    |                 |   |
+                    //              |    |e8               |e10|
+                    //           e11|    |                 |   |
+                    //              |    |_________________|___|
+                    //              |   / v0    e0         |   /v1
+                    //              |  /                   |  /
+                    //              | /e3                  | /e1
+                    //              |/_____________________|/
+                    //              v3         e2          v2
+                    //
+                    // the current voxel center is `v3`, the other vertices are centers of
+                    // the neighboring voxels
+                    let corners_absolute_coordinates = [
+                        Point3::new(x, y + 1, z),
+                        Point3::new(x + 1, y + 1, z),
+                        Point3::new(x + 1, y, z),
+                        Point3::new(x, y, z),
+                        Point3::new(x, y + 1, z + 1),
+                        Point3::new(x + 1, y + 1, z + 1),
+                        Point3::new(x + 1, y, z + 1),
+                        Point3::new(x, y, z + 1),
+                    ];
+                    let corners_cartesian_coordinates: Vec<_> = corners_absolute_coordinates
+                        .iter()
+                        .map(|a| absolute_voxel_to_cartesian_coordinate(a, &self.voxel_dimensions))
+                        .collect();
+
+                    // These points will become vertices of mesh patches forming
+                    // the outer envelope of the scalar field.
+                    #[rustfmt::skip]
+                    let edge_midpoints = vec![
```

Comment:
Use an array, no need to heap-allocate

Reply:
```
error[E0271]: type mismatch resolving `<std::slice::Iter<'_, nalgebra::geometry::point::Point<f32, nalgebra::base::dimension::U3>> as std::iter::IntoIterator>::Item == nalgebra::geometry::point::Point<f32, nalgebra::base::dimension::U3>`
   --> src\mesh\voxel_cloud.rs:635:25
    |
635 |                         Mesh::from_triangle_faces_with_vertices_and_computed_normals_remove_orphans(
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected reference, found struct `nalgebra::geometry::point::Point`
    | 
   ::: src\mesh\mod.rs:192:12
    |
192 |     pub fn from_triangle_faces_with_vertices_and_computed_normals_remove_orphans<F, V>(
    |            ---------------------------------------------------------------------
...
199 |         V: IntoIterator<Item = Point3<f32>>,
    |                         ------------------ required by this bound in `mesh::Mesh::from_triangle_faces_with_vertices_and_computed_normals_remove_orphans`
    |
    = note: expected type `&nalgebra::geometry::point::Point<f32, nalgebra::base::dimension::U3>`
               found type `nalgebra::geometry::point::Point<f32, 
nalgebra::base::dimension::U3>`
```

Reply:
Can't help you without more context.

Reply:
`edge_midpoints.iter().map(|v| *v)` ??

Reply:
If add the `.copied()` adapter, it should work.